### DISCUSSION TITLE
Inject different fetch_jwks function

### DISCUSSION THREAD

**Discussion by hannesvdvreken** [panva/jose#387] [Q&A] [answered]
Hi!

Is it possible to use a different HTTP client for `createRemoteJWKSet`? It's using the function in `fetch_jwks.ts` but that's using the default `http.get` method. I'd like to replace that with Axios, others might want to inject a different HTTP client.

My problem is that AWS ELB is blocking my requests made from `https.get` with a 403, whereas it's not blocking axios requests. I know this is not a JOSE problem but my own infrastructure problem but I have zero rights to change the ELB settings.

This should also help with https://github.com/panva/jose/discussions/240

**Comment by panva** [comment]
> Is it possible to use a different HTTP client for `createRemoteJWKSet`?

No.

> others might want to inject a different HTTP client.

But you can provide your own [`getKey`](https://github.com/panva/jose/blob/main/docs/functions/jwt_verify.jwtVerify.md#readme) key resolving function. I encourage you to publish your own key resolving functions as npm modules for others to discover and use if the default doesn't work for them.

> This should also help with https://github.com/panva/jose/discussions/240

There's nothing to do for #240, following redirects is not intended.

**Comment by hannesvdvreken** [comment, marked as answer]
Hi @panva,

Thank you very much for your answer.

The way I solved it was: 

```
import agentBase, { ClientRequest } from 'agent-base';

const agent = agentBase((req: ClientRequest, opts) => https.globalAgent);

$remote = createRemoteJWKSet(new URL(ssoJWKSurl), { agent });
```

This way I could tap into the http.Agent used by the fetch_jwks function. To follow the redirects (#240) this wouldn't work, but I suspect doing something with the agent would be the solution.

**Reply by panva** [reply]
So, as [documented](https://github.com/panva/jose/blob/main/docs/interfaces/jwks_remote.RemoteJWKSetOptions.md#agent).
